initial implementation of endpoints with fake data in UserController …

WorstWorm opened the pull request:
…class, UserDto class added, Lombok added

 On branch JDP230502-5
 Changes to be committed:
	modified:   build.gradle
	deleted:    src/main/java/com/kodilla/ecommercee/UserController.java
	new file:   src/main/java/com/kodilla/ecommercee/controller/UserController.java
	new file:   src/main/java/com/kodilla/ecommercee/dto/UserDto.java

WorstWorm committed:
initial implementation of endpoints with fake data in UserController class, UserDto class added, Lombok added

 On branch JDP230502-5
 Changes to be committed:
	modified:   build.gradle
	deleted:    src/main/java/com/kodilla/ecommercee/UserController.java
	new file:   src/main/java/com/kodilla/ecommercee/controller/UserController.java
	new file:   src/main/java/com/kodilla/ecommercee/dto/UserDto.java

JakubFir commented on src/main/java/com/kodilla/ecommercee/controller/UserController.java:
```diff
@@ -0,0 +1,102 @@
+
+@CrossOrigin("*")
+@RestController
+@RequestMapping("/user")
```
używałbym w request mappingu l.mn "/users", bedzie czytelniej w getMappingach według mnie.

WorstWorm replied:
Zgadzam się - poprawię.

JakubFir commented on src/main/java/com/kodilla/ecommercee/controller/UserController.java:
```diff
@@ -0,0 +1,102 @@
+    };
+    //=========================================
+
+    @GetMapping("/getUsers")
```
wtedy tutaj pusty @GetMapping

WorstWorm replied:
Zgadzam się - poprawię.

JakubFir commented on src/main/java/com/kodilla/ecommercee/controller/UserController.java:
```diff
@@ -0,0 +1,102 @@
+        return ResponseEntity.ok(userDtoList);
+    }
+
+    @GetMapping("/getUser/{id}")
```
tutaj @GetMapping("/{id}")

WorstWorm replied:
Zgadzam się - poprawię.

JakubFir commented on src/main/java/com/kodilla/ecommercee/controller/UserController.java:
```diff
@@ -0,0 +1,102 @@
+        return ResponseEntity.notFound().build();
+    }
+
+    @DeleteMapping("/deleteUser/{id}")
```
wysyłałbym samo "/{id}", bez "deleteUser", tak samo w post/put mappingu

WorstWorm replied:
Zgadzam się - poprawię.

JakubFir commented on src/main/java/com/kodilla/ecommercee/controller/UserController.java:
```diff
@@ -0,0 +1,102 @@
+    }
+
+    @PutMapping("/generateKey/{id}")
+    public ResponseEntity<Integer> generateKey(@PathVariable("id") long id) {
```
zastanowił bym się czy potrzebny jest endpoint do generowania klucza

WorstWorm replied:
Mamy go w specyfikacji projektu. Wymagane enpointy dla users:
utworzenie użytkownika
zablokowanie użytkownika
wygenerowanie losowego klucza ważnego godzinę po podaniu poprawnych danych użytkownika

WorstWorm committed:
Changing the request mapping address as suggested in conversation from pull request. Swagger added.

 On branch JDP230502-5
 Changes to be committed:
	modified:   build.gradle
	modified:   src/main/java/com/kodilla/ecommercee/EcommerceeApplication.java
	modified:   src/main/java/com/kodilla/ecommercee/controller/UserController.java
